pinecone/client.go

```diff
@@ -1,12 +1,33 @@
 package pinecone
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/deepmap/oapi-codegen/v2/pkg/securityprovider"
+	"github.com/pinecone-io/go-pinecone/internal/gen/control"
+	"io"
+	"net/http"
+)
+
 type Client struct {
-	apiKey string
+	apiKey     string
+	restClient *control.Client
 }
 
-func NewClient(apiKey string) *Client {
-	c := Client{apiKey: apiKey}
-	return &c
+func NewClient(apiKey string) (*Client, error) {
```

Comment:
We will definitely want to expand the options supported by the client in the future (for example, for proxy and SSL configurations similar to what I've been implementing in python this week). So some thought about if there's a way to handle future optional configs without creating a breaking change might be worthwhile.

Reply:
Good callout. I've changed to an input struct which will allow for additional params in the future without breaking current usages.

4d4dc0c